pkg/controller/job/job_controller.go

```diff
@@ -818,23 +822,23 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (forget bool, rEr
 			complete = succeeded >= *job.Spec.Completions && active == 0
 		}
 		if complete {
-			finishedCondition = newCondition(batch.JobComplete, v1.ConditionTrue, "", "")
```

Comment:
instead, make `newCondition` a method of the `Controller` struct

Reply:
/Ack

Reply:
/Ack

Doing this also means I will have to pull `ensureJobConditionStatus` into the `jm` as well. Or I have to keep passing the clock around still. Will do that as well.

Reply:
Oops, this was an old idea. Better pass the clock instead.

Reply:
Done